src/java/org/apache/cassandra/security/ISslContextFactory.java

```diff
@@ -99,6 +99,13 @@ default boolean hasKeystore()
         return true;
     }
 
+    /**
+     * Returns if this factory uses outbound keystore.
+     *
+     * @return {@code true} by default unless the implementation overrides this
+     */
+    boolean hasOutboundKeystore();
```

Comment:
Can this default to false to avoid breaking existing implementations?

Reply:
@jyothsnakonisa I think Jon's comment is in the same line of thought I had when I was asking about backwards compatibility since the older implementation doesn't have a need for outbound key configurations.